tools/run_tests/artifacts/build_artifact_python.sh

```diff
@@ -22,8 +22,17 @@ export PYTHON=${PYTHON:-python}
 export PIP=${PIP:-pip}
 export AUDITWHEEL=${AUDITWHEEL:-auditwheel}
 
-# Install Cython to avoid source wheel build failure.
-"${PIP}" install --upgrade cython
+if [ "$GRPC_SKIP_PIP_CYTHON_UPGRADE" == "" ]
+then
+  # Install Cython to avoid source wheel build failure.
+  # This only needs to be done when not running under docker (=on MacOS)
+  # since the docker images used for building python wheels
+  # already have a new-enough version of cython pre-installed.
+  # Any installation step is a potential source of breakages,
+  # so we are trying to perform as few download-and-install operations
+  # as possible.
+  "${PIP}" install --upgrade cython
```

Comment:
optional: We could pin the cython version to `0.29.21` as our `requirements.txt`.

Reply:
I'm going to leave as is and we can revisit this later.  Ideally I'd like avoid needing this step at all.
If I pinned here, the tricky part would be future maintenance - maintaining the pin being the same version in many different files is a bit tricky.